### DISCUSSION TITLE
[Feedback] <NextSeo /> with new app directory

### DISCUSSION THREAD

**Discussion by garmeeh** [garmeeh/next-seo#1073] [Q&A]
As the new app directory is still in beta, it would be great to keep any `NextSeo` related feedback here. 

Please check out the latest docs on how to use `NextSeo` with the app directory [here](https://github.com/garmeeh/next-seo/blob/master/APP_DIRECTORY.md)

If you have any feedback or find any bugs relating to Next.js be sure to check out their [feedback](https://github.com/vercel/next.js/discussions/41745) thread to see if it has been addressed or report your findings there if not.

**Comment by pzi** [comment]
Hey @garmeeh, I was wondering if the release containing the Next@13 changes should've been a major release instead of a minor one given a lot of the behaviour changes between these version as you called it out in your extra docs file.
This way, anyone just upgrading next-seo won't get a hefty surprise if things are suddenly different. Or does this behaviour only kick in when using the `useAppDir` prop?

Apart from that, the link to the app directory docs in the OG post is invalid.

**Reply by garmeeh** [reply]
Hey @pzi great question! 

So if you were to upgrade `next-seo` to the latest version when not using the app directory there will be no impact. 

The behaviour changes will only kick in when using the app directory. And for NextSeo to work in the new app directory you will need to add the `useAppDir` prop. 

The actual behaviour changes are a result of how Next.js now handles the `<head>` internally, which NextSeo had to adapt for.

**Reply by pzi** [reply]
Ok, the way I read the docs I thought it would change underlying stuff for everyone. All good then, no need for a major version then :)

**Comment by JamesSingleton** [comment]
I think it would be nice to use `<Script />` from `next/script` for the `application/ld+json`. It will be added to the bottom of the `<body />` tag but this way you can keep a majority of that stuff in the `head.tsx`. Furthermore, you would be able to add `application/ld+json` content in `app/layout.tsx` that would carry across ALL pages like I was doing with `<OrganizationJsonLd />` but now I can't do that.
